### DISCUSSION TITLE
Can we assume/require C99 or higher?

### DISCUSSION THREAD

**Discussion by mseminatore** [OpenMathLib/OpenBLAS#4364] [Q&A] [answered]
I am not sure what the min C compiler support level is for OpenBLAS. I know that C11 is not allowed as a requirement. But can we safely assume C99? In particular I want to know if it is OK to use variadic macros as in the following example:

```
#ifdef DEBUG
#define TRACE(...) printf(__VA_ARGS__)
#endif
```

**Comment by martin-frbg** [comment, marked as answer]
I think it should be alright in a MS Windows context, but I'd be more cautious about its use in common code (people stuck with ancient "enterprise" installations of Linux or even dinosaurs running SunOS etc)
